src/app/store/effects/VehicleTechRecordModel.effects.spec.ts

```diff
@@ -1,177 +1,260 @@
-import { forkJoin, Observable, of, ReplaySubject } from 'rxjs';
-import { VehicleTechRecordModelEffects } from '@app/store/effects/VehicleTechRecordModel.effects';
-import { TechnicalRecordServiceMock } from '../../../../testconfig/services-mocks/technical-record-service.mock';
 import { TestBed } from '@angular/core/testing';
-import { TechnicalRecordService } from '@app/technical-record-search/technical-record.service';
 import { RouterTestingModule } from '@angular/router/testing';
-import { INITIAL_STATE, Store } from '@ngrx/store';
-import { cold, hot } from 'jasmine-marbles';
+import { Router } from '@angular/router';
+import { Observable, BehaviorSubject } from 'rxjs';
+import { Action, Store } from '@ngrx/store';
 import { provideMockActions } from '@ngrx/effects/testing';
+import { hot, cold } from 'jasmine-marbles';
+
+import { MockStore } from '@app/utils';
+import { VehicleTechRecordModelEffects } from '@app/store/effects/VehicleTechRecordModel.effects';
+import { TechnicalRecordService } from '@app/technical-record-search/technical-record.service';
+import { UserService } from '@app/app-user.service';
 import {
-  GetVehicleTechRecordModelHavingStatusAll,
-  GetVehicleTechRecordModelHavingStatusAllSuccess,
-  SetVehicleTechRecordModelOnCreate
+  GetVehicleTechRecordHavingStatusAll,
+  GetVehicleTechRecordHavingStatusAllSuccess,
+  SetSelectedVehicleTechnicalRecord,
+  SetSelectedVehicleTechRecordSuccess,
+  SetViewState,
+  CreateVehicleTechRecord,
+  UpdateVehicleTechRecord,
+  UpdateVehicleTechRecordSuccess
 } from '@app/store/actions/VehicleTechRecordModel.actions';
-import { VehicleTechRecordModel } from '@app/models/vehicle-tech-record.model';
-import { addMatchers } from 'jasmine-marbles';
-import { IAppState } from '@app/store/state/app.state';
-import { VEHICLE_TECH_RECORD_SEARCH_ERRORS } from '@app/app.enums';
+import { GetVehicleTestResultModel } from '../actions/VehicleTestResultModel.actions';
+import { TechRecord } from './../../models/tech-record.model';
+import {
+  VehicleTechRecordModel,
+  VehicleTechRecordEdit
+} from '@app/models/vehicle-tech-record.model';
+import { VEHICLE_TECH_RECORD_SEARCH_ERRORS, VIEW_STATE } from '@app/app.enums';
 import { SearchParams } from '@app/models/search-params';
 import { SetErrorMessage } from '@app/store/actions/Error.actions';
 
-const techRecordModel: VehicleTechRecordModel = {
-  systemNumber: '1231243',
-  vrms: null,
-  vin: 'ABCDEFGH777777',
-  techRecord: [],
-  metadata: { adrDetails: undefined }
+const mockVehicleTechRecord = (): VehicleTechRecordModel => {
```

Comment:
Is that something we may want to be moving into our test util file?
We already know that it should be used as well by Test result.